Curve-finance-suite/src/agents/pool.migration.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+} from 'forta-agent';
+
+import {
+  provideFunctionCallsDetectorAgent,
+  FindingGenerator,
+} from 'nethermindeth-general-agents-module';
+
+export const MIGRATE_POOL_SIG = 'migrate_to_new_pool(address,address,uint256)';
+
+const createFindingGenerator = (alertId: string): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined): Finding => {
+    return Finding.fromObject({
+      name: 'Pool Migration Finding',
+      description: 'Pool migrated to new address',
+      alertId: alertId,
+      severity: FindingSeverity.Medium,
+      type: FindingType.Unknown,
+      metadata,
+    });
+  };
+};
+
+export default function provideMıgratePoolAgent(
+  alertID: string,
+  contractAddress: string
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const agentHandler = provideFunctionCallsDetectorAgent(
+      createFindingGenerator(alertID),
+      MIGRATE_POOL_SIG,
+      { to: contractAddress }
+    );
```

Comment:
You can create the `agentHandler` outside the `TransactionHandler` in order to create it just one time
```suggestion
   const agentHandler = provideFunctionCallsDetectorAgent(
    createFindingGenerator(alertID),
    MIGRATE_POOL_SIG,
    { to: contractAddress },
  );
  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
```